### PR TITLE
add macropad

### PR DESCRIPTION
**Pull request by zahranetme**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/zahranetme/macropad/8macropad/rules.mk [hunk 1]
```diff
@@ -0,0 +1,10 @@
+BOOTMAGIC_ENABLE = no
+MOUSEKEY_ENABLE = no
+EXTRAKEY_ENABLE = yes
+CONSOLE_ENABLE = yes
+COMMAND_ENABLE = yes
+NKRO_ENABLE = no
+BACKLIGHT_ENABLE = yes
+RGBLIGHT_ENABLE = yes
+AUDIO_ENABLE = no
+ENCODER_ENABLE = yes
```
Comment (adophoxia) [inline keyboards/zahranetme/macropad/8macropad/rules.mk:10]:
```suggestion
# This file intentionally left blank
```
Should be moved to `info.json`.

File: keyboards/zahranetme/macropad/keymaps/default/config.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#ifdef RGB_MATRIX_ENABLE
+#    define RGB_MATRIX_KEYPRESSES
+#endif
```
Comment (adophoxia) [inline keyboards/zahranetme/macropad/keymaps/default/config.h:20]: Is there a reason for this existing even though this board doesn't use RGB Matrix, and instead `rgblight` and/or `backlight`?

Edit: Realized this board actually has per-key LEDs on it. Did you forget to add RGB Matrix functionality for this board?

File: keyboards/zahranetme/macropad/keymaps/default/rules.mk [hunk 3]
```diff
@@ -0,0 +1,8 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+MOUSEKEY_ENABLE = yes
+CONSOLE_ENABLE = yes
+COMMAND_ENABLE = no
+ENCODER_MAP_ENABLE = yes
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes
```
Comment (adophoxia) [inline keyboards/zahranetme/macropad/keymaps/default/rules.mk:8]:
```suggestion
ENCODER_MAP_ENABLE = yes
```

File: keyboards/zahranetme/macropad/keymaps/via/rules.mk [hunk 4]
```diff
@@ -0,0 +1,8 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+MOUSEKEY_ENABLE = yes
+CONSOLE_ENABLE = yes
+COMMAND_ENABLE = no
+ENCODER_MAP_ENABLE = yes
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes
```
Comment (adophoxia) [inline keyboards/zahranetme/macropad/keymaps/via/rules.mk:8]:
```suggestion
VIA_ENABLE = yes
ENCODER_MAP_ENABLE = yes
```

**Review by adophoxia — CHANGES_REQUESTED** [review]

File: keyboards/zahranetme/macropad/keymaps/default/config.h (hunk 2 above)
Comment (zahranetme) [reply keyboards/zahranetme/macropad/keymaps/default/config.h:20]: it should be rgblight, what needs to be changed?

Comment (adophoxia) [reply keyboards/zahranetme/macropad/keymaps/default/config.h:20]: If this board doesn't actually use RGB Matrix despite the per-key LEDs shown in `readme.md` and `rgblight` instead, then you should remove this file and delete any code pertaining to `backlight` in this PR.

Comment (zahranetme) [reply keyboards/zahranetme/macropad/keymaps/default/config.h:20]: I only need led underglow

File: keyboards/zahranetme/macropad/8macropad/8macropad.c [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        }
+    } else if (index == 2) {
+        if (clockwise) {
+            tap_code(KC_NO);
+        } else {
+            tap_code(KC_NO);
+        }
+    }
+    return false;
+}
```
Comment (waffle87) [inline keyboards/zahranetme/macropad/8macropad/8macropad.c:60]:
```suggestion
```

File: keyboards/zahranetme/macropad/8macropad/config.h
Comment (waffle87) [inline keyboards/zahranetme/macropad/8macropad/config.h:1]: This file should be removed, as the functionality is not used anywhere.

File: keyboards/zahranetme/macropad/8macropad/info.json [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            "rgb_test": true,
+            "alternating": true,
+            "twinkle": true
+        }
+    },
+    "ws2812": {
+        "pin": "D3"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
```
Comment (waffle87) [inline keyboards/zahranetme/macropad/8macropad/info.json:43]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/zahranetme/macropad/8macropad/rules.mk [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes
```
Comment (waffle87) [inline keyboards/zahranetme/macropad/8macropad/rules.mk:13]:
```suggestion
# This file intentionally left blank
```

File: keyboards/zahranetme/macropad/8macropad/info.json [hunk 8]
```diff
@@ -0,0 +1,51 @@
+{
+    "keyboard_name": "ZahraNetMe 8MacroPad",
+    "manufacturer": "veriyan",
+    "url": "https://github.com/zahranetme",
+    "maintainer": "veriyan",
+    "usb": {
+      "vid": "0x4154",
+      "pid": "0x5A58",
+      "device_version": "1.0.0"
+    },
```
Comment (waffle87) [inline keyboards/zahranetme/macropad/8macropad/info.json:10]:
```suggestion
    },
    "features": {
        "bootmagic": true,
        "extrakey": true,
        "mousekey": true,
        "encoder": true,
        "rgblight": true
    },
```

File: keyboards/zahranetme/macropad/keymaps/default/config.h
Comment (waffle87) [inline keyboards/zahranetme/macropad/keymaps/default/config.h:1]: This file should be removed.

File: keyboards/zahranetme/macropad/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_MAIN] = {ENCODER_CCW_CW(KC_NO, KC_NO), ENCODER_CCW_CW(KC_NO, KC_NO)},
+    [_FN1]  = {ENCODER_CCW_CW(KC_NO, KC_NO), ENCODER_CCW_CW(KC_NO, KC_NO)},
+    [_FN2]  = {ENCODER_CCW_CW(KC_NO, KC_NO), ENCODER_CCW_CW(_______, _______)},
+    [_FN3]  = {ENCODER_CCW_CW(_______, _______), ENCODER_CCW_CW(_______, _______)},
+};
```
Comment (waffle87) [inline keyboards/zahranetme/macropad/keymaps/default/keymap.c:39]: Please include some actual functionality for both the encoders and keymap. It is not very useful for the default behaviour to be nothing.

File: keyboards/zahranetme/macropad/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_MAIN] = {ENCODER_CCW_CW(KC_NO, KC_NO), ENCODER_CCW_CW(KC_NO, KC_NO)},
+    [_FN1]  = {ENCODER_CCW_CW(KC_NO, KC_NO), ENCODER_CCW_CW(KC_NO, KC_NO)},
+    [_FN2]  = {ENCODER_CCW_CW(KC_NO, KC_NO), ENCODER_CCW_CW(_______, _______)},
+    [_FN3]  = {ENCODER_CCW_CW(_______, _______), ENCODER_CCW_CW(_______, _______)},
+};
```
Comment (waffle87) [inline keyboards/zahranetme/macropad/keymaps/via/keymap.c:39]: As above.

File: keyboards/zahranetme/macropad/keymaps/via/rules.mk [hunk 11]
```diff
@@ -0,0 +1,6 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
+MOUSEKEY_ENABLE = yes
+CONSOLE_ENABLE = yes
+COMMAND_ENABLE = no
+ENCODER_MAP_ENABLE = yes
```
Comment (waffle87) [inline keyboards/zahranetme/macropad/keymaps/via/rules.mk:6]:
```suggestion
VIA_ENABLE = yes
ENCODER_MAP_ENABLE = yes
```

File: keyboards/zahranetme/macropad/readme.md [hunk 12]
```diff
@@ -0,0 +1,26 @@
+# 8MACROPAD
+
+![8MACROPAD](https://i.imgur.com/s4fdlaY.png)
```
Comment (waffle87) [inline keyboards/zahranetme/macropad/readme.md:3]:
```suggestion
![8MACROPAD](https://i.imgur.com/s4fdlaYh.png)
```

**Review by waffle87 — CHANGES_REQUESTED** [review]
This file structure doesn't really make sense. Do you intend to submit more 8 key macropads that will be compatible with the keymaps in the upper level `macropad` directory?

**Comment by zahranetme** [comment]
> This file structure doesn't really make sense. Do you intend to submit more 8 key macropads that will be compatible with the keymaps in the upper level `macropad` directory? 

After I think about it, maybe I'll cancel it I will try to make a new one with more switches and add 3 rotary knobs thanks for the advice and Sorry to bother

**Pull request closed without merging**